tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -452,25 +452,11 @@ def with_poetry(context: PipelineContext) -> Container:
     python_with_poetry = with_pip_packages(python_with_git, ["poetry"])
 
     poetry_cache: CacheVolume = context.dagger_client.cache_volume("poetry_cache")
-    poetry_with_cache = python_with_poetry.with_mounted_cache("/root/.cache/pypoetry", poetry_cache, sharing=CacheSharingMode.PRIVATE)
```

Comment:
I private mode the cache is not shared between pipelines, I think we can opt for SHARED so that different poetry project we run in different pipelines can share the same cache.